CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+cmake_minimum_required(VERSION 3.4)
+project(RdKafka)
+
+# Options. No 'RDKAFKA_' prefix to match old C++ code. {
+
+# This option doesn't affect build in fact, only C code
+# (see 'rd_kafka_version_str'). In CMake the build type feature usually used
+# (like Debug, Release, etc.).
+option(WITHOUT_OPTIMIZATION "Disable optimization" OFF)
+
+option(ENABLE_DEVEL "Enable development asserts, checks, etc" OFF)
+option(ENABLE_REFCNT_DEBUG "Enable refcnt debugging" OFF)
+option(ENABLE_SHAREDPTR_DEBUG "Enable sharedptr debugging" OFF)
+
+option(WITH_ZLIB "With ZLIB" ON)
```

Comment:
Does this one fits your requirements?

| User's WITH_foo | User's ENABLE_foo | Package found? | Effective       |
| --------------- | ----------------- | -------------- | --------------- |
| *not set*       | YES               | *no*           | WITH_foo = NO   |
| *not set*       | YES               | *yes*          | WITH_foo = YES  |
| YES             | YES               | *no*           | **FATAL_ERROR** |
| YES             | YES               | *yes*          | WITH_foo = YES  |
| *not set*       | NO                | *x*            | WITH_foo = NO   |